manim/utils/hashing.py

```diff
@@ -0,0 +1,191 @@
+import json
+import zlib
+import inspect
+import copy
+import dis
+import numpy as np
+from types import ModuleType
+
+from ..logger import logger
+
+
+class CustomEncoder(json.JSONEncoder):
+    def default(self, obj):
+        """
+        This method is used to serialize objects to JSON format.
+        
+        If obj is a function, then it will return a dict with two keys : 'code', for the code source, and 'nonlocals' for all nonlocalsvalues. (including nonlocals functions, that will be serialized as this is recursive.)
+        if obj is a np.darray, it converts it into a list.
+        if obj is an object with __dict__ attribute, it returns its __dict__.
+        Else, will let the JSONEncoder do the stuff, and throw an error if the type is not suitable for JSONEncoder. 
+
+        Parameters
+        ----------
+        obj : `any`
```

Comment:
just   Any  should be ok
```suggestion
        obj : Any
```